server/src/internalClusterTest/java/org/elasticsearch/readiness/ReadinessClusterIT.java

```diff
@@ -319,4 +325,12 @@ public void testReadyAfterCorrectFileSettings() throws Exception {
         ReadinessService s = internalCluster().getInstance(ReadinessService.class, internalCluster().getMasterName());
         tcpReadinessProbeTrue(s);
     }
+
+    private void causeClusterStateUpdate() throws ExecutionException, InterruptedException {
+        ClusterUpdateSettingsRequest req = new ClusterUpdateSettingsRequest().persistentSettings(
+            Settings.builder().put("search.allow_expensive_queries", "false")
+        );
+        // This should succeed, it's never a reserved state in this test
+        client().admin().cluster().updateSettings(req).get();
```

Comment:
I'd rather we just explicitly did a cluster state update that doesn't change anything, if that's the intention here:

```suggestion
        internalCluster().getCurrentMasterNodeInstance(ClusterService.class)
            .submitUnbatchedStateUpdateTask("poke", new ClusterStateUpdateTask() {
                @Override
                public ClusterState execute(ClusterState currentState) {
                    return ClusterState.builder(currentState).build();
                }

                @Override
                public void onFailure(Exception e) {
                    assert false : e;
                }
            });
```

Reply:
NB edit^ (first suggestion was faulty)

Reply:
OK, sounds good. I tested this new approach with manually inserting a delay to cause the original issue, it works well now.